src/twister2/fixtures/common.py

```diff
@@ -0,0 +1,79 @@
+"""
+Module contains common code for all fixtures.
+"""
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+
+import pytest
+
+from twister2.exceptions import TwisterConfigurationException
+from twister2.platform_specification import PlatformSpecification
+from twister2.twister_config import TwisterConfig
+from twister2.yaml_test_specification import YamlTestSpecification
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class State:
+    should_run: bool
+    message: str = ''  # message for logging
+    reason: str = ''  # skip reason
+
+    def __bool__(self) -> bool:
+        return self.should_run
+
+
+class TestSetupManager:
```

Comment:
What do you think about preparing a fixture_setup_manager?
Than you run TestSetupManager initialization only once (now is called in fixture_builder, dut and log_parser).
In TestSetupManager is a lot of useful information about setup state, config etc. That fixture could be used also in test scenarios.

Reply:
I changed the name of this class to `TestSetupManager`. What name for a fixture would you like to have? `test_setup_manager` seems to be too long. Is `setup_manager` fine for you?

Reply:
For me `setup_maager` name will be ok